### PR TITLE
chore: Dependecy updates, such as kube 0.95.0 and Kubernetes 0.31

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

I was **not** able to update opentelemetry, and would let it to the experts to figure it out, CC @Techassi, @NickLarsenNZ and @nightkr 

I hope the `aws-lc-rs` crate will not cause any problems with Nix. I bumped zookeeper-operator and `make run-dev` worked fine.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by Techassi** [comment]
> I hope the `aws-lc-rs` crate will not cause any problems with Nix. I bumped zookeeper-operator and `make run-dev` worked fine.

We could evaluate if we can use `ring` as a crypt backend instead (supported by rustls).

**Comment by sbernauer** [comment]
> We could evaluate if we can use ring as a crypt backend instead (supported by rustls).

I sadly know the situation from trino-lb :/ This PR should not change anything (in comparison to main), and the zookeeper test worked, so I'd call it ready to be merged

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

File: crates/stackable-operator/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ All notable changes to this project will be documented in this file.
 
 - BREAKING: `validation` module now uses typed errors ([#851]).
 - Set `checkIncrement` to 5 seconds in Logback config ([#853]).
+- Bump to `kube` 0.95.0 and Kubernetes 1.31 ([#867]).
```
Comment (Techassi) [inline crates/stackable-operator/CHANGELOG.md:16]: suggestion: This PR not only bumps `kube`, but other dependencies as well.

```suggestion
- Bump Rust dependencies and enable Kubernetes 1.31 (via `kube` 0.95.0) ([#867]).
```

File: crates/stackable-operator/src/builder/pod/mod.rs [hunk 2]
```diff
@@ -456,7 +456,7 @@ impl PodBuilder {
     ) -> &mut Self {
         self.image_pull_secrets
             .get_or_insert_with(Vec::new)
-            .extend(secrets.map(|s| LocalObjectReference { name: Some(s) }));
+            .extend(secrets.map(|s| LocalObjectReference { name: s }));
```
Comment (Techassi) [inline crates/stackable-operator/src/builder/pod/mod.rs:459]: nit:

```suggestion
            .extend(secrets.map(|name| LocalObjectReference { name }));
```

**Review by Techassi — CHANGES_REQUESTED** [review]
LGTM, two small comments.

**Review by Techassi — APPROVED** [review]

**Pull request merged**